File: Problems/No.69_Sqrt(x)/sqrt(x).cpp
class Solution {
public:
    int mySqrt(int x) {
        
        // store some parameters
        int left = 0;
        int right = x;
        long mid;
        long Sqrt_ans;
        long temp;
        
        // start compute (BinarySearch method)
        while (left <= right) {
            
            mid = (left+right)/2;
            temp = mid*mid;
            
            if (temp == x) return mid;
            else if (temp > x) right = mid - 1;
            else {
                
                left = mid + 1;
                Sqrt_ans = mid;
                
            }
            
        }
        
        return Sqrt_ans;
        
    }
};

File: Problems/No.69_Sqrt(x)/sqrt(x).py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        
        # store some parameters
        left, right, mid = 0, x, None
        square = None
        sqrt_ans = None
        
        # start compute (BinarySearch method)
        while (left <= right):
            
            mid = (left+right)//2
            square = mid * mid;
            
            if (square == x):
                return mid
            elif (square > x):
                right = mid - 1
            else:
                left = mid + 1
                sqrt_ans = mid
            
        return sqrt_ans
